scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py

```diff
@@ -51,8 +58,8 @@ def compare_solutions(A, B, m):
     """Check eig vs. lobpcg consistency.
     """
     n = A.shape[0]
-    rnd = np.random.RandomState(0)
-    V = rnd.random((n, m))
+    rng = np.random.default_rng(0)
+    V = rng.random((n, m))
```

Comment:
Please use longer seeds for all default_rng's, see [here](http://scipy.github.io/devdocs/dev/contributor/rendering_documentation.html#examples-and-randomness)

Reply:
I think this is an old code, so making changes there would break the backward compatibility of the tests? If this is OK, I would be happy to update all such instances to meet the most recent standards.

Reply:
Strictly speaking, that's about user-facing examples, not tests. Generating arbitrary data from pseudorandomness in tests has no particular rules.

Actually, I would revert all of these back to `RandomState`; using it to generate test data is precisely `RandomState`'s remaining use case. We update `Generator`'s algorithms every so often, so it is not guaranteed to return the same data when numpy upgrades; `RandomState` does have that guarantee.

Reply:
I have no knowledge and thus no opinion myself on this matter. I hope that @rkern and @h-vetinari come to an agreement which I would be happy to implement.

Reply:
"backward compatibility of the tests" is something that makes no sense to me. We can update the tests however we see fit, including updating the expected results when we change a seed (though this test doesn't even seem to rely on specific values, so I'm not even sure that a different seed would break the test?)

> Actually, I would revert all of these back to `RandomState`

It's fine for me to leave things as-is; my point was: if we update to the new rng machinery[^1], we should do it properly.

[^1]: and achieving this consistently everywhere would have value IMO, i.e. no need to think about `RandomState` anymore.

> We update `Generator`'s algorithms every so often, so it is not guaranteed to return the same data when numpy upgrades;

Wouldn't that affect us (and other projects making heavy use of `Generator`s in their test suites) anyway? E.g. the specific results in https://github.com/scipy/scipy/commit/00f4e2e2f61769910b1b28f65871f82e17d7f8cc would also change if `Generator` were to change?

Reply:
There are two different things: APIs that intrinsically use PRNGs and creating test data to feed into APIs that don't. For the former (which is not the case with the PR, near as I can tell), yes, we'd like to move those APIs to primarily using `Generator`s and we'd want to test them using `Generator`s and not `RandomState`s. And yes, those tests will be susceptible to occasional changes but that's because the APIs are susceptible to those changes.

For the latter case of creating test data (which _is_ the relevant case for this PR), we had [extensive discussions culminating in NEP 19](https://numpy.org/neps/nep-0019-rng-policy.html#supporting-unit-tests) where we decided that a frozen `RandomState` _was_ desired to create stable, eternally reproducible test data so that unit tests wouldn't churn unnecessarily. We've already decided that we'll keep it around forever, so there's no benefit from dropping it in the one place it's actually useful.

Reply:
Thanks for the explanation. Sounds like we should just keep the tests as is.

Reply:
Resolved?

Reply:
Yes, we agree on reverting the tests back to using `RandomState`.

Reply:
done